p2p/net/swarm/dial_sync_test.go

```diff
@@ -52,16 +66,14 @@ func TestBasicDialSync(t *testing.T) {
 		finished <- struct{}{}
 	}()
 
-	// short sleep just to make sure we've moved around in the scheduler
-	time.Sleep(time.Millisecond * 20)
-	done()
-
-	<-finished
-	<-finished
+	// wait until both requests are registered
+	require.Eventually(t, func() bool { return len(requests) == 2 }, 100*time.Millisecond, time.Millisecond, "expected both requests to be processed")
+	// make the dials return
+	close(done)
+	// make sure the Dial functions return
+	require.Eventually(t, func() bool { return len(finished) == 2 }, 100*time.Millisecond, time.Millisecond, "dial functions should have returned")
```

Comment:
It does seem like the receive from the channel avoids the notion of time which is nice.